File: Practice 8/Кузнецов Михаил/accessing-data-jpa/src/main/java/com/example/accessingdatajpa/model/Reader.java
package com.example.accessingdatajpa.model;

import javax.persistence.*;
import java.util.Date;

@Entity
@Table
public class Reader {
    @Id
    @GeneratedValue(strategy= GenerationType.AUTO)
    @Column
    private Long id;
    @Column
    private String lastName;
    @Column
    private String firstName;
    @Column
    private String address;
    @Column
    private Date birthDate;
}

File: Practice 8/PopovOO/libraryBD/src/main/kotlin/ru/hse/librarybd/dto/Reader.kt
package ru.hse.librarybd.dto

import java.sql.Date
import javax.persistence.Entity
import javax.persistence.GeneratedValue
import javax.persistence.Id
import javax.persistence.OneToMany

@Entity(name = "readers")
data class Reader(
    @Id @GeneratedValue
    val readerId: Long = -1,
    val lastName: String,
    val firstName: String,
    val address: String,
    val dateOfBirth: Date,
    @OneToMany
    val borrowings: List<Borrowing> = mutableListOf()
) {
    constructor() : this(-1, "", "", "", Date(0), mutableListOf())
}
